optimum/intel/openvino/quantization.py

```diff
@@ -318,80 +321,95 @@ def _quantize_ovbasemodel(
         self,
         ov_config: OVConfig,
         save_directory: Union[str, Path] = None,
-        calibration_dataset: Optional[Union[datasets.Dataset, nncf.Dataset, Iterable]] = None,
+        calibration_dataset: Optional[Union["Dataset", nncf.Dataset, Iterable]] = None,
         batch_size: int = 1,
         data_collator: Optional[DataCollator] = None,
         remove_unused_columns: bool = True,
         **kwargs,
     ):
+        from optimum.intel.openvino.modeling_diffusion import OVStableDiffusionPipelineBase
+
         if save_directory is not None:
             save_directory = Path(save_directory)
             save_directory.mkdir(parents=True, exist_ok=True)
-
         quantization_config = ov_config.quantization_config
+
+        if calibration_dataset is not None:
+            # Process custom calibration dataset
+
+            if isinstance(self.model, OVStableDiffusionPipelineBase):
+                calibration_dataset = self._prepare_unet_dataset(
+                    quantization_config.num_samples, dataset=calibration_dataset
+                )
+            elif Dataset is not None and isinstance(calibration_dataset, Dataset):
```

Comment:
Please correct the part `elif Dataset is not None`

Reply:
This part is actually correct. This is to adequately process the case when `datasets` is not available. There is such code above:
```
if is_datasets_available():
    from datasets import Dataset
else:
    Dataset = None
```